packages/e2e/test/wallet/SingleAddressWallet/delegation-rewards.test.ts

```diff
@@ -0,0 +1,156 @@
+import * as envalid from 'envalid';
+import { Cardano, createSlotEpochCalc } from '@cardano-sdk/core';
+import { SignedTx, SingleAddressWallet, buildTx } from '@cardano-sdk/wallet';
+import { TestWallet, getWallet } from '../../../src';
+import { assertTxIsValid, waitForWalletStateSettle } from '../../../../wallet/test/util';
+import { filter, firstValueFrom } from 'rxjs';
+import { logger } from '@cardano-sdk/util-dev';
+import { requestCoins, runningAgainstLocalNetwork, submitAndConfirm, transferCoins, waitForEpoch } from '../util';
+
+// Verify environment.
+export const env = envalid.cleanEnv(process.env, {
```

Comment:
Let's turn this into a function to DRY the code, as there's quite a bit of duplication already.

something like this in the `util` package?:
```ts
const cleanEnv = (requiredProviders: Providers[], env: NodeJS.Env) => {
  const specs = {}
  if (requiredProviders.contains(Providers.Asset)) {
    specs.ASSET_PROVIDER = envalid.str(),
    specs.ASSET_PROVIDER_PARAMS = envalid.json({ default: {} }),
  }
  ...
  return envalid.cleanEnv(env, specs)
}
```

Reply:
Captured in ADP-2334